Firestore/core/src/firebase/firestore/model/field_value.h

```diff
@@ -289,6 +290,23 @@ class ObjectValue : public util::Comparable<ObjectValue> {
   FieldValue fv_;
 };
 
+class FieldValue::ServerTimestamp {
+ public:
+  ServerTimestamp(Timestamp local_write_time, FSTFieldValue* previous_value);
+
+  explicit ServerTimestamp(Timestamp local_write_time);
```

Comment:
Optional: use a single constructor with a default argument `=nil` for `previous_value`? I _think_ `explicit` should work for that case as well.

Reply:
Done.